File: Java/src/main/java/QuestionBank/Solution001_100/Solution35.java
package QuestionBank.Solution001_100;

/**
 * @author windows
 */
public class Solution35 {

    public int searchInsert(int[] nums, int target) {
        int left = - 1;
        int right = nums.length;
        int middle = -1;
        while(left != right - 1) {
            middle = (left + right) / 2;
            if (nums[middle] == target) {
                return middle;
            }else if (nums[middle] < target) {
                left = middle;
            }else{
                right = middle;
            }
        }
        return right;
    }

    public static void main(String[] args) {
        Solution35 solution35 = new Solution35();
        System.out.println(solution35.searchInsert(new int[]{1, 3, 5, 6}, 2));
    }
}

File: C++/src/Solution1_100/Solution35.cpp
//
// Created by windows on 2023/3/1.
//

#include "Solution35.h"
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

int Solution35::searchInsert(vector<int> &nums, int target) {
    int left = 0;
    int right = nums.size() - 1;
    while (left <= right) {
        int mid = left + (right - left) / 2;
        if (nums[mid] == target) {
            return mid;
        }else if (nums[mid] < target) {
            left = mid + 1;
        } else{
            right = mid - 1;
        }
    }
    return left;
}
